pipelines/pipeline.yml

```diff
@@ -184,37 +138,160 @@ stages:
                     goPath: "$(System.DefaultWorkingDirectory)/gopath"
                     goBin: "$(System.DefaultWorkingDirectory)/gopath/bin"
                   displayName: "Set up the Go workspace"
-
                 - bash: |
-                    cp gomod/go.mod .
-                    cp gosum/go.sum .
+                    set -ex
                     echo '##vso[task.prependpath]$(GOBIN)'
                     echo '##vso[task.prependpath]$(GOROOT)/bin'
                     go get github.com/jstemmer/go-junit-report
+
+                    # Download dce-cli
+                    wget -q https://github.com/Optum/dce-cli/releases/download/v0.3.0/dce_linux_amd64.zip
+                    # Validate chechsum
+                    expected_sha="d19e1ecee5eb9646dbd5bf509ea813f4db1c22bf843551bd1ec8c8f42efe54c2"
+                    test $(shasum -a256 ./dce_linux_amd64.zip | awk '{print $1}') == "${expected_sha}"
+                    unzip ./dce_linux_amd64.zip -d ./
+
+                    pwd
+                    ls
+
                   displayName: "Setup Go"
-                  workingDirectory: "$(Pipeline.Workspace)"
 
-                # Functional Tests
+                # Lease a DCE account, to use for deploying our PR environment
+                # (deploy DCE in DCE)
                 - task: AWSShellScript@1
-                  displayName: "Functional Tests"
+                  displayName: "Lease DCE Account"
                   inputs:
-                    awsCredentials: "AWS_Login"
+                    awsCredentials: "AWS_DCE_API_NONPROD"
                     regionName: "us-east-1"
                     scriptType: "inline"
                     inlineScript: |
-                      set -euxo pipefail
-                      cd $(Pipeline.Workspace)
-                      mkdir -p junit-report
-                      # Run functional tests
-                      go test -v ./tests/... 2>&1 | tee >(go-junit-report > junit-report/functional.xml)
-
-                # Publish junit test results (for unit and functional tests)
-                - task: PublishTestResults@2
+                      set -ex
+
+                      echo "Principal ID is ${NAMESPACE}"
+
+                      echo "Configuring the dce-cli"
+                      echo "
+                        api:
+                          # nonprod (0086)
+                          host: $(DCE_NONPROD_HOSTNAME)
+                          basepath: /api
+                        region: us-east-1
+                      " > ./dce.yml
+
+
+                      # Check if we already have an active lease for this PR
```

Comment:
Half this bash code could deleted if we fixed a couple bugs in the dce-cli 
(outputing to stderr, invalid JSON, unsupported `GET` query params, etc)